src/renderer/components/NewConnectionForm.tsx

```diff
@@ -0,0 +1,247 @@
+
+interface INewConnectionForm {
+  nickname: string;
+  type: DBProvider;
+  address: string;
+  port: string;
+  username: string;
+  password: string;
+  remember: boolean;
+}
+
+interface INewConnectionErrors {
+  nickname?: string;
+  type?: string;
+  address?: string;
+  port?: string;
+  username?: string;
+  password?: string;
+}
+
+const NewConnectionForm = () => {
+  const defaultForm: INewConnectionForm = {
+    nickname: '',
+    type: DBProvider.PostgreSQL,
+    address: '',
+    port: '',
+    username: '',
+    password: '',
+    remember: false,
+  };
+
+  const [form, setForm] = useState<INewConnectionForm>(defaultForm);
+  const [errors, setErrors] = useState<INewConnectionErrors>({});
+  const [alert, setAlert] = useState<boolean>(false);
+
+  const navigate = useNavigate();
+
+  const setField = (field: string, value: string | DBProvider | boolean) => {
+    setForm({
+      ...form,
+      [field]: value,
+    });
+
+    if (errors[field as keyof INewConnectionErrors])
+      setErrors({
+        ...errors,
+        [field]: null,
+      });
+  };
+
+  const clearForm = () => {
+    setForm(defaultForm);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const enumVal = event.target.value as DBProvider;
+    setField('type', enumVal);
+  };
+
+  const validateForm = () => {
+    const newErrors: INewConnectionErrors = {};
+    const { nickname, type, address, port, username, password } = form;
+    if (!nickname) {
+      newErrors.nickname = 'Please type a nickname.';
+    }
+    if (!type) {
+      newErrors.type = 'Please select a database type.';
+    }
+    if (!address) {
+      newErrors.address = 'Please type an address.';
+    }
+    if (!port) {
+      newErrors.port = 'Please type a port.';
+    } else if (Number.isNaN(parseInt(port, 10))) {
+      newErrors.port = 'Port must be a number.';
+    }
+    if (!username) {
+      newErrors.username = 'Please type a database user.';
+    }
+    if (!password) {
+      newErrors.password = 'Please type a password.';
+    }
+
+    return newErrors;
+  };
+  const handleSubmit = async (event: React.FormEvent<HTMLElement>) => {
+    event.preventDefault();
+
+    const formErrors = validateForm();
+    if (Object.keys(formErrors).length > 0) {
+      setErrors(formErrors);
+      return;
+    }
+
+    const connection: ConnectionModelType = {
+      nickname: form.nickname,
+      type: form.type,
+      connectionConfig: {
+        config: 'manual',
+        address: form.address,
+        port: parseInt(form.port, 10),
+        username: form.username,
+        password: form.password,
+      },
+    };
+
+    try {
+      // creates the server connection in sqlite and connects to it
+      await window.connections.ipcRenderer.create(connection);
+
+      // check if server exists by attempting to fetch procedures - might be a better way to do this
+      await window.procedures.ipcRenderer.fetchProcedures();
```

Comment:
@adityavidyadharan what is the best way to check that the server exists? the create function does not seem to return any errors for garbage data